### ISSUE TITLE
RECOMMEND/VOTE Papers

### ISSUE THREAD

**Issue by sxjscience** [ML-HK/paper-discussion-group#3]
# How to recommend

We can recommend some papers for further discussion under this issue. Include a link to the paper + the conference name and other related information (like the abstract, some basic descriptions, links to samples code or online demonstrations).

Please only include one topic per comment. For example, if you propose to discuss "paper X" which is heavily based on "paper Y" and you believe both have to be read together (possibly over multiple weeks) just create one comment for that. If you propose two unrelated papers please create two comments.

For example, the following markdown format could be used.

``` markdown
[**PAPER-NAME**](PAPER-LINK) (AUTHORS)
(CONFERENCE/JOURNAL)
_ABSTRACT_ 
```
# How to vote

Please vote using the "Thumbs up" emoji :thumbsup:. Some papers will be marked as **Discussed** and you should only vote for the undiscussed papers.

**Comment by sxjscience** [comment]
By @leezu 

[**Training Very Deep Networks**](https://papers.nips.cc/paper/5850-training-very-deep-networks.pdf) (Rupesh Kumar Srivastava, Klaus Greff, Jürgen Schmidhuber)
(NIPS 2015)
_Theoretical and empirical evidence indicates that the depth of neural networks is crucial for their success. However, training becomes more difficult as depth increases, and training of very deep networks remains an open problem. Here we introduce a new architecture designed to overcome this. Our so-called highway networks allow unimpeded information flow across many layers on information highways. They are inspired by Long Short-Term Memory recurrent networks and use adaptive gating units to regulate the information flow. Even with hundreds of layers, highway networks can be trained directly through simple gradient descent. This enables the study of extremely deep and efficient architectures._

**Comment by sxjscience** [comment]
By @leezu 

[**On the difficulty of training recurrent neural networks**](http://jmlr.org/proceedings/papers/v28/pascanu13.pdf) (Razvan Pascanu, Tomas Mikolov, Yoshua Bengio)
(ICML2013 )
_There are two widely known issues with properly training recurrent neural networks, the vanishing and the exploding gradient problems detailed in Bengio et al. (1994). In this paper we attempt to improve the understanding of the underlying issues by exploring these problems from an analytical, a geometric and a dynamical systems perspective. Our analysis is used to justify a simple yet effective solution. We propose a gradient norm clipping strategy to deal with exploding gradients and a soft constraint for the vanishing gradients problem. We validate empirically our hypothesis and proposed solutions in the experimental section._

**Comment by sxjscience** [comment]
By @sxjscience

[**Dueling Network Architectures for Deep Reinforcement Learning**](http://jmlr.org/proceedings/papers/v48/wangf16.pdf) (Ziyu Wang, Tom Schaul, Matteo Hessel, Hado van Hasselt, Marc Lanctot, Nando de Freitas)
(ICML 2016)
_In recent years there have been many successes of using deep representations in reinforcement learning. Still, many of these applications use conventional architectures, such as convolutional networks, LSTMs, or auto-encoders. In this paper, we present a new neural network architecture for model-free reinforcement learning. Our dueling network represents two separate estimators: one for the state value function and one for the state-dependent action advantage function. The main benefit of this factoring is to generalize learning across actions without imposing any change to the underlying reinforcement learning algorithm. Our results show that this architecture leads to better policy evaluation in the presence of many similar-valued actions. Moreover, the dueling architecture enables our RL agent to outperform the state-of-the-art on the Atari 2600 domain._

**Comment by sxjscience** [comment]
By @zyq40137 (**6th Session**)

[**Variational Autoencoder for Deep Learning of Images, Labels and Captions**](http://people.ee.duke.edu/~lcarin/Yunchen_nips_2016.pdf) (Yunchen Pu, Zhe Gan, Ricardo Henao, Xin Yuan, Chunyuan Li, Andrew Stevens and Lawrence Carin)
(NIPS 2016)
_A novel variational autoencoder is developed to model images, as well as associated
labels or captions. The Deep Generative Deconvolutional Network (DGDN) is used
as a decoder of the latent image features, and a deep Convolutional Neural Network
(CNN) is used as an image encoder; the CNN is used to approximate a distribution
for the latent DGDN features/code. The latent code is also linked to generative
models for labels (Bayesian support vector machine) or captions (recurrent neural
network). When predicting a label/caption for a new image at test, averaging is
performed across the distribution of latent codes; this is computationally efficient as
a consequence of the learned CNN-based encoder. Since the framework is capable
of modeling the image in the presence/absence of associated labels/captions, a
new semi-supervised setting is manifested for CNN learning with images; the
framework even allows unsupervised CNN learning, based on images alone._

**Comment by sxjscience** [comment]
By @yyuanad (**4th Session**)

[**InfoGAN: Interpretable Representation Learning by Information Maximizing Generative Adversarial Nets**](https://arxiv.org/pdf/1606.03657v1.pdf) (Xi Chen, Yan Duan, Rein Houthooft, John Schulman, Ilya Sutskever, Pieter Abbeel)
(NIPS2016)
_This paper describes InfoGAN, an information-theoretic extension to the Generative
Adversarial Network that is able to learn disentangled representations in a
completely unsupervised manner. InfoGAN is a generative adversarial network
that also maximizes the mutual information between a small subset of the latent
variables and the observation. We derive a lower bound of the mutual information
objective that can be optimized efficiently. Specifically, InfoGAN successfully
disentangles writing styles from digit shapes on the MNIST dataset, pose from
lighting of 3D rendered images, and background digits from the central digit on
the SVHN dataset. It also discovers visual concepts that include hair styles, presence/absence
of eyeglasses, and emotions on the CelebA face dataset. Experiments
show that InfoGAN learns interpretable representations that are competitive with
representations learned by existing supervised methods._

**Comment by sxjscience** [comment]
By @yyuanad 

[**Harnessing Deep Neural Networks with Logic Rules**](http://www.cs.cmu.edu/~epxing/papers/2016/Hu_etal_ACL16.pdf)(Zhiting Hu, Xuezhe Ma, Zhengzhong Liu, Eduard Hovy, Eric P. Xing)
(ACL2016)(one of the 10 outstanding papers)
_Combining deep neural networks with structured logic rules is desirable to harness flexibility and reduce uninterpretability of the neural models. We propose a general framework capable of enhancing various types of neural networks (e.g., CNNs and RNNs) with declarative first-order logic rules. Specifically, we develop an iterative distillation method that transfers the structured
information of logic rules into the weights of neural networks. We deploy the framework on a CNN for sentiment analysis, and an RNN for named entity recognition. With a few highly intuitive rules, we obtain substantial improvements and achieve state-of-the-art or comparable results to previous best-performing systems._

**Comment by sxjscience** [comment]
[**Trust Region Policy Optimization**](https://arxiv.org/pdf/1502.05477v4.pdf) (John Schulman, Sergey Levine, Philipp Moritz, Michael Jordan, Pieter Abbeel )
(ICML2015)
_We describe a iterative procedure for optimizing policies, with guaranteed monotonic improvement. By making several approximations to the theoretically-justified procedure, we develop a practical algorithm, called Trust Region Policy Optimization (TRPO). This algorithm is similar to natural policy gradient methods and is effective for optimizing large nonlinear policies such as neural networks. Our experiments demonstrate its robust performance on a wide variety of tasks: learning simulated robotic swimming, hopping, and walking gaits; and playing Atari games using images of the screen as input. Despite its approximations that deviate from the theory, TRPO tends to give monotonic improvement, with little tuning of hyperparameters._

**Comment by sxjscience** [comment]
By @sxjscience (7th session)
[**Hybrid computing using a neural network with dynamic external memory**](http://www.nature.com/nature/journal/vaop/ncurrent/full/nature20101.html) (Alex Graves, Greg Wayne, Malcolm Reynolds,   Tim Harley, Ivo Danihelka,  Agnieszka Grabska-Barwińska,   Sergio Gómez Colmenarejo,  Edward Grefenstette,    Tiago Ramalho, John Agapiou,    Adrià Puigdomènech Badia, Karl Moritz Hermann,    Yori Zwols, Georg Ostrovski,    Adam Cain,  Helen King, Christopher Summerfield,    Phil Blunsom,   Koray Kavukcuoglu   & Demis Hassabis)
(Nature 2016)
_Artificial neural networks are remarkably adept at sensory processing, sequence learning and reinforcement learning, but are limited in their ability to represent variables and data structures and to store data over long timescales, owing to the lack of an external memory. Here we introduce a machine learning model called a differentiable neural computer (DNC), which consists of a neural network that can read from and write to an external memory matrix, analogous to the random-access memory in a conventional computer. Like a conventional computer, it can use its memory to represent and manipulate complex data structures, but, like a neural network, it can learn to do so from data. When trained with supervised learning, we demonstrate that a DNC can successfully answer synthetic questions designed to emulate reasoning and inference problems in natural language. We show that it can learn tasks such as finding the shortest path between specified points and inferring the missing links in randomly generated graphs, and then generalize these tasks to specific graphs such as transport networks and family trees. When trained with reinforcement learning, a DNC can complete a moving blocks puzzle in which changing goals are specified by sequences of symbols. Taken together, our results demonstrate that DNCs have the capacity to solve complex, structured tasks that are inaccessible to neural networks without external read–write memory._

**Comment by sxjscience** [comment]
[**Stein Variational Gradient Descent: A General Purpose Bayesian Inference Algorithm**](https://arxiv.org/pdf/1608.04471v2.pdf) (Qiang Liu, Dilin Wang)
(NIPS 2016)
_We propose a general purpose variational inference algorithm that forms a natural
counterpart of gradient descent for optimization. Our method iteratively transports
a set of particles to match the target distribution, by applying a form of
functional gradient descent that minimizes the KL divergence. Empirical studies
are performed on various real world models and datasets, on which our method is
competitive with existing state-of-the-art methods. The derivation of our method
is based on a new theoretical result that connects the derivative of KL divergence
under smooth transforms with Stein’s identity and a recently proposed kernelized
Stein discrepancy, which is of independent interest._

**Comment by peterzcc** [comment]
[Strategic Attentive Writer for Learning Macro-Actions
](https://arxiv.org/abs/1606.04695)
(Alexander (Sasha) Vezhnevets, Volodymyr Mnih, John Agapiou, Simon Osindero, Alex Graves, Oriol Vinyals, Koray Kavukcuoglu)
We present a novel deep recurrent neural network architecture that learns to build implicit plans in an end-to-end manner by purely interacting with an environment in reinforcement learning setting. The network builds an internal plan, which is continuously updated upon observation of the next input from the environment. It can also partition this internal representation into contiguous sub- sequences by learning for how long the plan can be committed to - i.e. followed without re-planing. Combining these properties, the proposed model, dubbed STRategic Attentive Writer (STRAW) can learn high-level, temporally abstracted macro- actions of varying lengths that are solely learnt from data without any prior information. These macro-actions enable both structured exploration and economic computation. We experimentally demonstrate that STRAW delivers strong improvements on several ATARI games by employing temporally extended planning strategies (e.g. Ms. Pacman and Frostbite). It is at the same time a general algorithm that can be applied on any sequence data. To that end, we also show that when trained on text prediction task, STRAW naturally predicts frequent n-grams (instead of macro-actions), demonstrating the generality of the approach.

**Comment by yyuanad** [comment]
By @sxjscience (7th session)
[Neural Turing Machines](https://arxiv.org/pdf/1410.5401v2.pdf)(Alex Graves, Greg Wayne, Ivo Danihelka)(DeepMind)

We extend the capabilities of neural networks by coupling them to external memory resources,
which they can interact with by attentional processes. The combined system is
analogous to a Turing Machine or Von Neumann architecture but is differentiable end-to-end,
allowing it to be efficiently trained with gradient descent. Preliminary results demonstrate
that Neural Turing Machines can infer simple algorithms such as copying, sorting,
and associative recall from input and output examples.

**Comment by sxjscience** [comment]
[**Value Iteration Networks**](https://arxiv.org/pdf/1602.02867v2.pdf) (Aviv Tamar, Yi Wu, Garrett Thomas, Sergey Levine, and Pieter Abbeel)
(NIPS 2016)
_We introduce the value iteration network (VIN): a fully differentiable neural network
with a ‘planning module’ embedded within. VINs can learn to plan, and are
suitable for predicting outcomes that involve planning-based reasoning, such as
policies for reinforcement learning. Key to our approach is a novel differentiable
approximation of the value-iteration algorithm, which can be represented as a convolutional
neural network, and trained end-to-end using standard backpropagation.
We evaluate VIN based policies on discrete and continuous path-planning domains,
and on a natural-language based search task. We show that by learning an explicit
planning computation, VIN policies generalize better to new, unseen domains._

**Comment by WellyZhang** [comment]
By @WellyZhang (8th session)
[You Only Look Once: Unified, Real-Time Object Detection](https://arxiv.org/abs/1506.02640) (Joseph Redmon, Santosh Divvala, Ross Girshick, Ali Farhadi)
(CVPR 2016 Oral)
*We present YOLO, a new approach to object detection. Prior work on object detection repurposes classifiers to perform detection. Instead, we frame object detection as a regression problem to spatially separated bounding boxes and associated class probabilities. A single neural network predicts bounding boxes and class probabilities directly from full images in one evaluation. Since the whole detection pipeline is a single network, it can be optimized end-to-end directly on detection performance. Our unified architecture is extremely fast. Our base YOLO model processes images in real-time at 45 frames per second. A smaller version of the network, Fast YOLO, processes an astounding 155 frames per second while still achieving double the mAP of other real-time detectors. Compared to state-of-the-art detection systems, YOLO makes more localization errors but is less likely to predict false positives on background. Finally, YOLO learns very general representations of objects. It outperforms other detection methods, including DPM and R-CNN, when generalizing from natural images to other domains like artwork.*

[SSD: Single Shot MultiBox Detector](https://arxiv.org/abs/1512.02325)(Wei Liu, Dragomir Anguelov, Dumitru Erhan, Christian Szegedy, Scott Reed, Cheng-Yang Fu, Alexander C. Berg)
(ECCV 2016 Oral)
*We present a method for detecting objects in images using a single deep neural network. Our approach, named SSD, discretizes the output space of bounding boxes into a set of default boxes over different aspect ratios and scales per feature map location. At prediction time, the network generates scores for the presence of each object category in each default box and produces adjustments to the box to better match the object shape. Additionally, the network combines predictions from multiple feature maps with different resolutions to naturally handle objects of various sizes. Our SSD model is simple relative to methods that require object proposals because it completely eliminates proposal generation and subsequent pixel or feature resampling stage and encapsulates all computation in a single network. This makes SSD easy to train and straightforward to integrate into systems that require a detection component. Experimental results on the PASCAL VOC, MS COCO, and ILSVRC datasets confirm that SSD has comparable accuracy to methods that utilize an additional object proposal step and is much faster, while providing a unified framework for both training and inference. Compared to other single stage methods, SSD has much better accuracy, even with a smaller input image size. For 300×300 input, SSD achieves 72.1% mAP on VOC2007 test at 58 FPS on a Nvidia Titan X and for 500×500 input, SSD achieves 75.1% mAP, outperforming a comparable state of the art Faster R-CNN model.*
